System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -174,45 +171,55 @@ public override string[] GetFiles(string path, string searchPattern)
         }
 
         public override string[] GetFiles(string path, string searchPattern, SearchOption searchOption)
+        {
+            return GetFilesInternal(mockFileDataAccessor.AllFiles, path, searchPattern, searchOption);
+        }
+
+        private string[] GetFilesInternal(
+            IEnumerable<string> files,
+            string path,
+            string searchPattern,
+            SearchOption searchOption)
         {
             if (path == null)
             {
                 throw new ArgumentNullException(nameof(path));
             }
-            
+
             if (path.Any(c => Path.GetInvalidPathChars().Contains(c)))
             {
                 throw new ArgumentException("Invalid character(s) in path", nameof(path));
             }
 
+            CheckSearchPattern(searchPattern);
+            path = path.TrimSlashes();
+            path = path.NormalizeSlashes();
+
             if (!Exists(path))
             {
-                throw new DirectoryNotFoundException(string.Format(CultureInfo.InvariantCulture, StringResources.Manager.GetString("COULD_NOT_FIND_PART_OF_PATH_EXCEPTION"), path));
+                throw new DirectoryNotFoundException(
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        StringResources.Manager.GetString("COULD_NOT_FIND_PART_OF_PATH_EXCEPTION"),
+                        path));
             }
 
-            return GetFilesInternal(mockFileDataAccessor.AllFiles, path, searchPattern, searchOption);
-        }
-
-        private string[] GetFilesInternal(IEnumerable<string> files, string path, string searchPattern, SearchOption searchOption)
-        {
-            CheckSearchPattern(searchPattern);
-            path = path.TrimSlashes();
             path = EnsureAbsolutePath(path);
 
-            if (!path.EndsWith(Path.DirectorySeparatorChar.ToString())
-                && !path.EndsWith(Path.AltDirectorySeparatorChar.ToString()))
+            if (!path.EndsWith(Path.DirectorySeparatorChar.ToString()))
             {
                 path += Path.DirectorySeparatorChar;
             }
 
-            bool isUnix = XFS.IsUnixPlatform();
+            var isUnix = XFS.IsUnixPlatform();
 
-            string allDirectoriesPattern = isUnix
+            var allDirectoriesPattern = isUnix
```

Comment:
I personally don't like "var"-ing everything unless the type is explicitly clear.   In this case, you have to look up isUnix to find the type, and its been changed to var...

Reply:
`isUnix` isn't being assigned to the new variable, it's the condition in a ternary (`... ? ... : ...`) which picks between two string literals.